File: apple/Sources/ReactNativeWatchOS/Modifiers/GaugeStyle.swift
import SwiftUI

/// SwiftUI `.gaugeStyle(_:)`. Maps the `style` string to a gauge style.
/// All listed styles are available on watchOS 9 (the deployment target),
/// so no availability gating is required. Unknown values fall back to
/// `.automatic`.
enum RNWGaugeStyleModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("gaugeStyle") { view, params, _ in
            switch params.string("style") {
            case "accessoryCircular":
                return AnyView(view.gaugeStyle(.accessoryCircular))
            case "accessoryCircularCapacity":
                return AnyView(view.gaugeStyle(.accessoryCircularCapacity))
            case "accessoryLinear":
                return AnyView(view.gaugeStyle(.accessoryLinear))
            case "accessoryLinearCapacity":
                return AnyView(view.gaugeStyle(.accessoryLinearCapacity))
            case "linearCapacity":
                return AnyView(view.gaugeStyle(.linearCapacity))
            default:
                return AnyView(view.gaugeStyle(.automatic))
            }
        }
    }
}

File: src/modifiers/gaugeStyle.ts
import { createModifier } from './createModifier';

export type GaugeStyle =
  | 'automatic'
  | 'accessoryCircular'
  | 'accessoryCircularCapacity'
  | 'accessoryLinear'
  | 'accessoryLinearCapacity'
  | 'linearCapacity';

/// SwiftUI `.gaugeStyle(_:)`. Sets the visual style applied to `Gauge`s
/// within the view. `'automatic'` defers to the platform default. Some
/// capacity styles require newer watchOS; unsupported values leave the
/// view unchanged on older systems.
export function gaugeStyle(style: GaugeStyle = 'automatic') {
  return createModifier('gaugeStyle', { style });
}
